### PR TITLE
Changes for Implementation of a Neural Network

### PR DESCRIPTION
**Pull request by shwetu-rohatgi**
Add basic code implementation of a neural network with an explanation of forward and backward propagation.

### CODE REVIEW

**Comment by pawan-mehta** [comment]
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  ossbackup2@gmail.com
    Domain oss-india.net has exceeded the max emails per hour (109/100 (109%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from mail-oi0-f50.google.com ([209.85.218.50]:44853)
	by server14.hostingraja.in with esmtps (TLSv1.2:ECDHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.89)
	(envelope-from <pawan.ossindia+caf_=pawan=oss-india.net@gmail.com>)
	id 1e5IiB-001KO4-SP
	for pawan@oss-india.net; Fri, 20 Oct 2017 03:20:57 +0530
Received: by mail-oi0-f50.google.com with SMTP id v132so17263610oie.1
        for <pawan@oss-india.net>; Thu, 19 Oct 2017 14:50:23 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:delivered-to:dkim-signature:dkim-signature:date
         :from:reply-to:to:cc:message-id:subject:mime-version
         :content-transfer-encoding:precedence:list-id:list-archive:list-post
         :list-unsubscribe;
        bh=V5/jDRt4bxW/f4fh7KWRRcCjrSerZ58VoXQDgU/vsy0=;
        b=KGqjzGFKcy7IuqIvT8YBMcD8z8kE3WIFb+1/dCQtG8dQx5ORACve8nduRNRBP9WYzt
         11uaI3AJ9T1vL7gGNaHhUfiz2OOLfx2IGNCB3NEyaqdsV8nvojAYYfMsWpLOn8m/IshF
         ANlDkrmhgMs0dQXY86z6+t8y/QMIr4uQxPfQh8H/aj0TZFv68aj+xQ3AEr5NgR5rnMog
         2qN3EHMNza9dwRkkihksR+bWOp4W1iRAZo7ffZyQ5VSwqw4r/HFtecM0QH6Ch0Jye3yZ
         XcNl7sBu3uutc3XIUx38/806yie60HYOiIZqqPrCfvZOPRmSitHJHIqBhcEEED+FEdy5
         MEVA==
X-Gm-Message-State: AMCzsaUL8vk8V56n9lCwBtuwDSycZ+k1P6WXl8RrtAjPSBdPUfVS8Fe6
	zCFMlg149uLGlnHyZ4h7v7YrxQ6Iel9p3VF0OLPu6OSP2tYFBVw=
X-Received: by 10.157.33.59 with SMTP id i56mr1958511otb.248.1508449801120;
        Thu, 19 Oct 2017 14:50:01 -0700 (PDT)
X-Forwarded-To: pawan@oss-india.net
X-Forwarded-For: pawan.ossindia@gmail.com pawan@oss-india.net
Delivered-To: pawan.ossindia@gmail.com
Received: by 10.168.68.11 with SMTP id r11csp4355140ofa;
        Thu, 19 Oct 2017 14:50:00 -0700 (PDT)
X-Google-Smtp-Source: ABhQp+Q6ZlikNRKab8THeqP0wp6zkUxz+vGwSdYoM6JGCkvbssrjMBcdMP5H99wcKiHq1MyO/PUX
X-Received: by 10.36.69.222 with SMTP id c91mr4497707itd.125.1508449800423;
        Thu, 19 Oct 2017 14:50:00 -0700 (PDT)
ARC-Seal: i=1; a=rsa-sha256; t=1508449800; cv=none;
        d=google.com; s=arc-20160816;
        b=u0AlYd+KVV4r7VnUSGhg0yE1q3056EyA/3q/rCn/JswJLZeCxVTM67TwAt/ZTK3DUy
         5C3tPoUM293R/Rkt8ypOwloyKjgwia2a7nK6eua2jU3E9UdVczGdFYK7qPEV81VPMxPg
         mG8hyQwEjvxFK9Q1UYF+dPj9sp4S876xJqbRize2Ii+6nK6JxTxxF5b149Vddg4m+QqU
         youYZvfpM/MxBnSZ4/UsSpJQNK920Q5mdxfzYBPnDi7/jnWm1sTrOxbAi6HeAel5+fhQ
         bnxuxhzZBRJSBFslYHJIPzPdNktwRPvUsbJUPHV6rdxdmBGESLarkymGxWc109jASRlu
         UCxQ==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=list-unsubscribe:list-post:list-archive:list-id:precedence
         :content-transfer-encoding:mime-version:subject:message-id:cc:to
         :reply-to:from:date:dkim-signature:dkim-signature
         :arc-authentication-results;
        bh=V5/jDRt4bxW/f4fh7KWRRcCjrSerZ58VoXQDgU/vsy0=;
        b=LU8TEhlR1kJXxbpDdExZPxTuvnoSdyElKaeOru160+rWpzC8CVjFTDV6T+Qj5fW1sx
         wAV6H+4P8uWWTumgQxyAHUkVpHpRhXPa2VdJkKeAU2MMMFzx07Jdwrx/VtYhez3oeoJu
         JEnkM2RHc/Gr0MPbpvcx2iwbj3LWI7JGBvxoF0wOfnQOIfgX6IgIwhEAgmMhjoGx7pJ/
         33S+K+dYgjjM+K09VgabV1QH2S7PYcJ4JmOUP2bvditKi/MS4j13fxCizkzgfPcXgUKE
         DnSqvrHAqHj6ei6nP1QUYZ86ECnH8pbTFZSgKIGCWXxa4/0efjjp+VCmzDvKkkOWdFwQ
         xlKA==
ARC-Authentication-Results: i=1; mx.google.com;
       dkim=pass header.i=@github.com header.s=s20150108 header.b=ShZgICkv;
       dkim=pass header.i=@sendgrid.info header.s=smtpapi header.b=C4DvKh/U;
       spf=pass (google.com: domain of bounces+848413-3702-pawan.ossindia=gmail.com@sgmail.github.com designates 192.254.112.98 as permitted sender) smtp.mailfrom=bounces+848413-3702-pawan.ossindia=gmail.com@sgmail.github.com;
       dmarc=pass (p=NONE sp=NONE dis=NONE) header.from=github.com
Received: from o3.sgmail.github.com (o3.sgmail.github.com. [192.254.112.98])
        by mx.google.com with ESMTPS id o140si2061768ito.127.2017.10.19.14.50.00
        for <pawan.ossindia@gmail.com>
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Thu, 19 Oct 2017 14:50:00 -0700 (PDT)
Received-SPF: pass (google.com: domain of bounces+848413-3702-pawan.ossindia=gmail.com@sgmail.github.com designates 192.254.112.98 as permitted sender) client-ip=192.254.112.98;
Authentication-Results: mx.google.com;
       dkim=pass header.i=@github.com header.s=s20150108 header.b=ShZgICkv;
       dkim=pass header.i=@sendgrid.info header.s=smtpapi header.b=C4DvKh/U;
       spf=pass (google.com: domain of bounces+848413-3702-pawan.ossindia=gmail.com@sgmail.github.com designates 192.254.112.98 as permitted sender) smtp.mailfrom=bounces+848413-3702-pawan.ossindia=gmail.com@sgmail.github.com;
       dmarc=pass (p=NONE sp=NONE dis=NONE) header.from=github.com
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=github.com; 
	h=from:reply-to:to:cc:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=1E8PbMeIQPHx6nLdzw3r9eg8As8=; b=ShZgICkv0c5pUA1r
	fZ6yOJxs/mEm2KnFJNxEizOLKGX0vXDM3meR6DP8+9AV74c9T0D6L2qw8NKtDsq9
	3NDJJe5Mf5LvYXqinjgnd4824uCdHCvUctjkISDzYnXFuTEn2qjqdWyM67iR9a3Q
	7zj0AAfgh+dwgTmSOOZ2LuE8hH4=
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed/relaxed; d=sendgrid.info; 
	h=from:reply-to:to:cc:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe:x-feedback-id; 
	s=smtpapi; bh=1E8PbMeIQPHx6nLdzw3r9eg8As8=; b=C4DvKh/UjXjp5gzO36
	1484JysJabcZ7WjFB0HW4zzNMgQacSCCiXzmqk8JaggaNmmHVm1eicl7J3EUV1lR
	1v38piVeZLTrAHJTnY4BC9j43kpswYFT0z8zA+yR+E8eT/7dS22SWE4C4bOqsCEo
	vZvXS9WHend5413xLZ6EYN9MU=
Received: by filter0481p1mdw1.sendgrid.net with SMTP id filter0481p1mdw1-25706-59E91E08-C
        2017-10-19 21:50:00.25071111 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd0005p1iad1.sendgrid.net (SG) with ESMTP id ofHLmalMRmeRk3VKzfklYw
	for <pawan.ossindia@gmail.com>; Thu, 19 Oct 2017 21:50:00.214 +0000 (UTC)
Date: Thu, 19 Oct 2017 21:50:00 +0000 (UTC)

File: src/pages/machine-learning/neural-networks/index.md [hunk 1]
```diff
@@ -31,3 +31,81 @@ Neural networks are used on a variety of tasks. These include computer vision, s
 - <a href='http://ufldl.stanford.edu/tutorial/supervised/MultiLayerNeuralNetworks/' target='_blank' rel='nofollow'>Stanford University</a>
 - <a href='https://youtu.be/aircAruvnKk' target='_blank' rel='nofollow'>3Blue1Brown</a>
 - <a href='https://youtu.be/h3l4qz76JhQ' target='_blank' rel='nofollow'>Siraj Raval</a>
+
+My Changes  -----------------
```
Comment (jp-sauve) [inline src/pages/machine-learning/neural-networks/index.md]: "My Changes" should not be in this article. Same with "End". Your submission should be in the form that the final article should take.

File: src/pages/machine-learning/neural-networks/index.md [hunk 2]
```diff
@@ -31,3 +31,81 @@ Neural networks are used on a variety of tasks. These include computer vision, s
...
+
+The brain consists of hundreds of billion of cells called neurons. These neurons are connected together by synapses which are nothing but the connections across which a neuron can send an impulse to another neuron. When a neuron sends an excitatory signal to another neuron, then this signal will be added to all of the other inputs of that neuron. If it exceeds a given threshold then it will cause the target neuron to fire an action signal forward — this is how the thinking process works internally.
+
+In Computer Science, we model this process by creating “networks” on a computer using matrices. These networks can be understood as abstraction of neurons without all the biological complexities taken into account. To keep things simple, we will just model a simple NN, with two layers capable of solving linear classification problem.
+
+Forward Propagation:
```
Comment (jp-sauve) [inline src/pages/machine-learning/neural-networks/index.md]: This looks like a header. Check out https://github.com/freeCodeCamp/guides#article-style-guide for style guidelines.

File: src/pages/machine-learning/neural-networks/index.md [hunk 3]
```diff
@@ -31,3 +31,81 @@ Neural networks are used on a variety of tasks. These include computer vision, s
...
+Take the inputs, multiply by the weights (just use random numbers as weights)
+Let Y = WiIi = W1I1+W2I2+W3I3
+Pass the result through a sigmoid formula to calculate the neuron’s output. The Sigmoid function is used to normalise the result between 0 and 1:
+1/1 + e-y
+
+Back Propagation
```
Comment (jp-sauve) [inline src/pages/machine-learning/neural-networks/index.md]: Another header that needs style

File: src/pages/machine-learning/neural-networks/index.md [hunk 4]
```diff
@@ -31,3 +31,81 @@ Neural networks are used on a variety of tasks. These include computer vision, s
...
+    # The training set.
+    inputs = array([[0, 1, 1], [1, 0, 0], [1, 0, 1]])
+    outputs = array([[1, 0, 1]]).T
+ 
+    # Train the neural network
+    neural_network.train(inputs, outputs, 10000)
+ 
+    # Test the neural network with a test example.
+    print neural_network.learn(array([1, 0, 1]))
+
```
Comment (jp-sauve) [inline src/pages/machine-learning/neural-networks/index.md]: The preceding code block should be enclosed in markdown.

File: src/pages/machine-learning/neural-networks/index.md [hunk 5]
```diff
@@ -31,3 +31,81 @@ Neural networks are used on a variety of tasks. These include computer vision, s
...
+    # Train the neural network
+    neural_network.train(inputs, outputs, 10000)
+ 
+    # Test the neural network with a test example.
+    print neural_network.learn(array([1, 0, 1]))
+    
+Output - 
+After 10 iterations our neural network predicts the value to be 0.65980921.  It looks not good as the answer should really be 1. If we increase the number of iterations to 100, we get 0.87680541. Our network is getting smarter! Subsequently, for 10000 iterations we get 0.9897704 which is pretty close and indeed a satisfactory output.
+
+My changes end -----------------------
```
Comment (jp-sauve) [inline src/pages/machine-learning/neural-networks/index.md]: this isn't needed.

**Pull request closed without merging**